src/sqlfluff/utils/analysis/select.py

```diff
@@ -158,3 +161,47 @@ def _get_pivot_table_columns(segment, dialect):
             pivot_table_column_aliases.append(pivot_table_column_alias.raw)
 
     return pivot_table_column_aliases
+
+
+# Lambda arguments,
+# e.g. `x` and `y` in `x -> x is not null` and `(x, y) -> x + y`
+# are declared in-place, and are as such standalone – i.e. they do not reference
+# identifiers or columns that we should expect to be declared somewhere else.
+# These columns are interesting to identify since they can get special
+# treatment in some rules.
+def _get_lambda_argument_columns(segment, dialect):
+    if not dialect or dialect.name not in ["athena", "sparksql"]:
+        # Only athena and sparksql are known to have  lambda expressions,
+        # so all other dialects will have zero lambda columns
+        return []
+
+    lambda_argument_columns = []
+    for potential_lambda in segment.recursive_crawl("expression"):
+        if (
+            potential_lambda.get_child("binary_operator")
+            and potential_lambda.get_child("binary_operator").raw == "->"
+        ):
+            arrow_operator = potential_lambda.get_child("binary_operator")
```

Comment:
This can probably be walrussed, but black and I don't seem to agree on what that should look like, so I kept it simple with a pre-definition outside of the if-statement